core/src/main/resources/error/error-classes.json

```diff
@@ -17,20 +17,41 @@
   "CANNOT_GENERATE_CODE_FOR_EXPRESSION" : {
     "message" : [ "Cannot generate code for expression: %s" ]
   },
+  "CANNOT_GET_JDBC_TYPE" : {
+    "message" : [ "Can't get JDBC type for %s" ]
+  },
+  "CANNOT_OPERATE_ON_CANONICALIZATION_PLAN" : {
+    "message" : [ "Operating on canonicalization plan" ]
+  },
   "CANNOT_PARSE_DECIMAL" : {
     "message" : [ "Cannot parse decimal" ],
     "sqlState" : "42000"
   },
+  "CANNOT_READ_FOOTER_FOR_FILE" : {
+    "message" : [ "Could not read footer for file: %s" ],
+    "sqlState" : "42000"
+  },
   "CANNOT_TERMINATE_GENERATOR" : {
     "message" : [ "Cannot terminate expression: %s" ]
   },
+  "CANNOT_TRANSLATE_NON_NULL_VALUE_FOR_FIELD" : {
+    "message" : [ "Can't translate non-null value for field %s" ]
+  },
   "CAST_CAUSES_OVERFLOW" : {
     "message" : [ "Casting %s to %s causes overflow" ],
     "sqlState" : "22005"
   },
   "CONCURRENT_QUERY" : {
     "message" : [ "Another instance of this query was just started by a concurrent session." ]
   },
+  "DATA_TYPE_UNSUPPORTED" : {
+    "message" : [ "Data type '%s' is not supported." ],
+    "sqlState" : "0A000"
+  },
+  "DDL_UNSUPPORTED" : {
```

Comment:
To match the existing unsupported operation error class, can you rename this to `UNSUPPORTED_DDL`?